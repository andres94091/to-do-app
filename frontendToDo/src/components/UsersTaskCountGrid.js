import React, { useEffect, useState } from 'react'
import axios from 'axios'

const UsersTaskCountGrid = ({bandReload}) => {

  const [arrayUsersTasks, setArrayUsersTasks] = useState([])

  useEffect(async () => {
    const response = await axios.get(`http://localhost:5000/api/users`)
    const { data: usersData } = response.data
    setArrayUsersTasks(usersData)
  }, [bandReload])

  const renderHeader = () => {
    const header = ['Id', 'Name', 'Task per user']
    return header.map((column, index) => {
      return <th key={index}>{column}</th>
    })
  }

  const renderBody = () => {
    return arrayUsersTasks.map(({id, name, amounttask}) => {
      return (
        <tr key = {id}>
          <td>{id}</td>
          <td>{name}</td>
          <td>{amounttask}</td>
        </tr>
      )
    })
  }

  return (
    <>
      <table className="users-task-table table table-striped">
        <thead>
          <tr className= "users-task-table-header">{renderHeader()}</tr>
        </thead>
        <tbody>
          {renderBody()}
        </tbody>
      </table>
    </>
  )
}

export default UsersTaskCountGrid
