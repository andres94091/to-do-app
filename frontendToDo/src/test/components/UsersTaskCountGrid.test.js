import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import UsersTaskCountGrid from '../../components/UsersTaskCountGrid';

describe('tests in <UsersTaskCountGrid/>', () => {

  const wrapper = shallow(<UsersTaskCountGrid
    bandReload = {true}
  />)

  it('should render the component correctly', () => {

    expect(wrapper).toMatchSnapshot();
  })
  
  
})
