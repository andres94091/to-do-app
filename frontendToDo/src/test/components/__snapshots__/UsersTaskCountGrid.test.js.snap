// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <UsersTaskCountGrid/> should render the component correctly 1`] = `
<Fragment>
  <table
    className="users-task-table table table-striped"
  >
    <thead>
      <tr
        className="users-task-table-header"
      >
        <th
          key="0"
        >
          Id
        </th>
        <th
          key="1"
        >
          Name
        </th>
        <th
          key="2"
        >
          Task per user
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
